packages/launchpad-ledger/src/launchpadledger.ts

```diff
@@ -0,0 +1,232 @@
+    // ledger-cosmos-js hardens the first three indices
+    const response = await this.cosmosApp.sign(unharden(hdPathToUse), fromUtf8(message));
+    this.handleLedgerErrors(response, {
+      rejectionMessage: "Transaction signing request was rejected by the user",
+    });
+    return Secp256k1Signature.fromDer((response as SignResponse).signature).toFixedLength();
+  }
+
+  private verifyAppMode(testMode: boolean): void {
+    if (testMode && !this.testModeAllowed) {
+      throw new Error(`DANGER: The Cosmos Ledger app is in test mode and should not be used on mainnet!`);
+    }
+  }
+
+  private async getOpenAppName(): Promise<string> {
+    await this.connect();
+    assert(this.cosmosApp, "Cosmos Ledger App is not connected");
+
+    const response = await this.cosmosApp.appInfo();
+    this.handleLedgerErrors(response);
+    return (response as AppInfoResponse).appName;
+  }
+
+  private async verifyAppVersion(): Promise<void> {
+    const version = await this.getCosmosAppVersion();
+    if (!semver.gte(version, requiredCosmosAppVersion)) {
+      throw new Error("Outdated version: Please update Cosmos Ledger App to the latest version.");
+    }
+  }
+
+  private async verifyCosmosAppIsOpen(): Promise<void> {
+    const appName = await this.getOpenAppName();
+
+    if (appName.toLowerCase() === `dashboard`) {
+      throw new Error(`Please open the Cosmos Ledger app on your Ledger device.`);
+    }
+    if (appName.toLowerCase() !== `cosmos`) {
+      throw new Error(`Please close ${appName} and open the Cosmos Ledger app on your Ledger device.`);
+    }
+  }
+
+  private async verifyDeviceIsReady(): Promise<void> {
+    await this.verifyAppVersion();
+    await this.verifyCosmosAppIsOpen();
+  }
+
+  /* eslint-disable @typescript-eslint/naming-convention */
+  private handleLedgerErrors(
+    {
+      error_message: errorMessage,
+      device_locked: deviceLocked = false,
+    }: { error_message: string; device_locked?: boolean },
+    { rejectionMessage = "Request was rejected by the user" } = {},
+  ): void {
+    if (deviceLocked) {
+      throw new Error("Ledger’s screensaver mode is on");
+    }
+    switch (errorMessage) {
+      case "U2F: Timeout":
+        throw new Error("Connection timed out. Please try again.");
+      case "Cosmos app does not seem to be open":
+        throw new Error("Cosmos app is not open");
+      case "Command not allowed":
+        throw new Error("Transaction rejected");
+      case "Transaction rejected":
+        throw new Error(rejectionMessage);
+      case "Unknown Status Code: 26628":
+        throw new Error("Ledger’s screensaver mode is on");
+      case "Instruction not supported":
+        throw new Error(
+          `Your Cosmos Ledger App is not up to date. Please update to version ${requiredCosmosAppVersion}.`,
+        );
+      case "No errors":
+        break;
+      default:
+        throw new Error(`Ledger Native Error: ${errorMessage}`);
+    }
+  }
+  /* eslint-enable */
```

Comment:
Can we reduce the scope of this eslint-disable by writing some of

```ts
  private handleLedgerErrors(
    {
      error_message: errorMessage,
      device_locked: deviceLocked = false,
    }: { error_message: string; device_locked?: boolean },
    { rejectionMessage = "Request was rejected by the user" } = {},
  ): void {
```

as interfaces?

Reply:
Doesn't seem so 😞 
https://github.com/typescript-eslint/typescript-eslint/issues/2250#issuecomment-670055096